src/app/retrospective/retrospective-create/retrospective-create.component.ts

```diff
@@ -46,19 +62,96 @@ export class RetrospectiveCreateComponent implements OnInit, OnDestroy {
+        this.getTeamList();
         this.addTaskProvider();
+        if (this.isUpdateMode) {
+            this.hidePassword();
+            this.getFieldsEditLevel();
+        }
     }
 
     ngOnDestroy() {
         this.destroy$.next(true);
         this.destroy$.complete();
     }
+    get projectNameControl() {
+        return this.retroFormGroup.get('projectName');
+    }
+
+    hidePassword() {
+        if (this.retrospective.TaskProviderConfig[0].data.credentials.type === 'basicAuth') {
+            // Replacing orginal password With dummy value
+            this.originalPassword = this.retrospective.TaskProviderConfig[0].data.credentials.password;
+            this.retrospective.TaskProviderConfig[0].data.credentials.password = DUMMY_HIDDEN_VALUE;
+        } else if (this.retrospective.TaskProviderConfig[0].data.credentials.type === 'apiToken') {
+            // Replacing orginal API token With dummy value
+            this.originalPassword = this.retrospective.TaskProviderConfig[0].data.credentials.apiToken;
+            this.retrospective.TaskProviderConfig[0].data.credentials.apiToken = DUMMY_HIDDEN_VALUE;
+        }
+    }
+
+    getFieldsEditLevel() {
+        this.retrospectiveService.getFieldEditLevel(this.retrospectiveID).subscribe(
+            response => {
+                this.fieldsEditableMap = response.data;
+                this.setValueAndDisableState('title', 'Title');
+                this.setValueAndDisableState('team', 'TeamID');
+                this.setValueAndDisableState('storyPointPerWeek', 'StoryPointPerWeek');
+                this.setValueAndDisableState('projectName', 'ProjectName');
+                this.projectNameControl.value.split(',').forEach((element: string) => {
+                    this.ProjectNames.set(element, false);
+                });
+                this.isRetrospectLoaded = true;
+            },
+          );
+    }
+
+    getEditableState(name: string): boolean {
+        return !(this.isUpdateMode && this.fieldsEditableMap[name] === EDIT_LEVELS.NOT_EDITABLE);
+    }
+
+    // This function will set value and its edit states of form fields.
+    setValueAndDisableState(formFieldName: string, name: string) {
+        this.retroFormGroup.patchValue({[formFieldName]: this.retrospective[name]});
+
+        if (this.fieldsEditableMap[name] === EDIT_LEVELS.NOT_EDITABLE) {
+            this.retroFormGroup.get(formFieldName).disable();
+        }
+    }
+
+    getProjectMap() {
+        return Array.from(this.ProjectNames.keys());
+    }
+    addChip(event: MatChipInputEvent) {
+        const input = event.input;
+        const value = event.value;
+
+        // Reset the input value
+      if (input) { input.value = ''; }
+
+      if (Array.from(this.ProjectNames.keys()).findIndex(element => value.toLowerCase() === element.toLowerCase()) > -1) {
+        return;
+      }
+      if ((value || '').trim()) {
+            // Add our project
+            this.ProjectNames.set(value.trim(), true);
```

Comment:
Why not trim the value in the first statement?

Reply:
Done.